src/utils/MainApi.js

```diff
@@ -0,0 +1,70 @@
+import { checkResponse } from "./Api";
+export const baseUrl =
+  process.env.NODE_ENV === "production"
+    ? "https://api.newsapp.crabdance.com"
+    : "http://localhost:3000";
+
+const headers = {
+  "Content-type": "application/json",
+};
+
+export const signUp = ({ name, email, password }) => {
+  return fetch(`${baseUrl}/signup`, {
+    method: "POST",
+    headers: headers,
+    body: JSON.stringify({ email, password, name }),
+  }).then(checkResponse);
```

Comment:
**Could be improved**

You can make a special method (function) for fetching and checking responses **not to duplicate** it in every request:

```js

   function request(url, options) {
    return fetch(url, options).then(checkResponse)
  }

```

And now you can safely replace `fetch` with `request` and remove repeated checking. And you don’t need to change anything else.